packages/ag-charts-community/src/chart/mapping/types.ts

```diff
@@ -42,19 +42,18 @@ export type SeriesOptionsTypes =
 
 export type SeriesType = SeriesOptionsTypes['type'];
 
-export function optionsType(input: { series?: { type?: SeriesType }[] }): NonNullable<SeriesType> {
-    return input.series?.[0]?.type ?? 'line';
+export function optionsType(input: { series?: { type?: SeriesType }[] }): SeriesType {
+    const { series } = input;
+    if (series) {
+        return input.series?.[0]?.type ?? 'line';
+    }
+    return undefined;
```

Comment:
```suggestion
    if (series) {
        return series?.[0]?.type ?? 'line';
    }
```
nit: cleaner + I'm on the side that thinks `undefined` shouldn't be used as a value 😄

Reply:
My version:

```suggestion
    if (!series) return;
    return series[0]?.type ?? 'line';
```

Reply:
Thanks both, was meaning to clean this up.